orderbook/src/bin/orderbook.rs

```diff
@@ -0,0 +1,64 @@
+use actix::Actor;
+use actix_web::middleware;
+use actix_web::web;
+use actix_web::App;
+use actix_web::HttpServer;
+use diesel::r2d2;
+use diesel::r2d2::ConnectionManager;
+use diesel::PgConnection;
+use orderbook::routes::delete_order;
+use orderbook::routes::get_order;
+use orderbook::routes::get_orders;
+use orderbook::routes::index;
+use orderbook::routes::post_order;
+use orderbook::routes::price_feed_route;
+use orderbook::routes::update_order;
+use orderbook::routes::websocket;
+use orderbook::run_migration;
+use std::sync::atomic::AtomicUsize;
+use std::sync::Arc;
+use tracing::level_filters::LevelFilter;
+
+#[actix_web::main]
+async fn main() -> std::io::Result<()> {
+    orderbook::logger::init_tracing(LevelFilter::DEBUG, false).unwrap();
+
+    // set up database connection pool
+    let conn_spec = "postgres://postgres:mysecretpassword@localhost:5432/orderbook".to_string();
+    let manager = ConnectionManager::<PgConnection>::new(conn_spec);
+    let pool = r2d2::Pool::builder()
+        .build(manager)
+        .expect("Failed to create pool.");
+
+    let mut conn = pool.get().unwrap();
+    run_migration(&mut conn);
+
+    println!("starting HTTP server at http://localhost:8000");
+
+    // set up applications state
+    // keep a count of the number of visitors
+    let app_state = Arc::new(AtomicUsize::new(0));
+
+    // start price feed server actor
+    let ws_server = websocket::PriceFeed::new(app_state.clone(), pool.clone()).start();
+
+    // Start HTTP server
+    HttpServer::new(move || {
+        App::new()
+            // set up DB pool to be used with web::Data<Pool> extractor
+            .app_data(web::Data::new(pool.clone()))
+            .wrap(middleware::Logger::default())
```

Comment:
I wonder if there is any tracing middleware for actix :thinking: if we want to go that route again, we should consider ahead of time how much detail we want to include in the traces so we avoid explosive disk usage

Reply:
Looks like there is a lib for it. 
I agree, better look into this sooner than later :)